UI/NativeApp.cpp

```diff
@@ -482,27 +495,44 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 	g_Config.defaultCurrentDirectory = Path("/");
 	g_Config.internalDataDirectory = Path(savegame_dir);
 
-#if defined(__ANDROID__)
-	// TODO: This needs to change in Android 12.
-	//
-	// Maybe there should be an option to use internal memory instead, but I think
-	// that for most people, using external memory (SDCard/USB Storage) makes the
-	// most sense.
-	g_Config.defaultCurrentDirectory = Path(external_dir);
+#if PPSSPP_PLATFORM(ANDROID)
+	// In Android 12 with scoped storage, due to the above, the external directory
+	// is no longer the plain root of external storage, but it's an app specific directory
+	// on external storage (g_extFilesDir).
+	if (System_GetPropertyBool(SYSPROP_ANDROID_SCOPED_STORAGE)) {
+		g_Config.defaultCurrentDirectory = Path(g_extFilesDir);
+	} else {
+		// Maybe there should be an option to use internal memory instead, but I think
+		// that for most people, using external memory (SDCard/USB Storage) makes the
+		// most sense.
+		g_Config.defaultCurrentDirectory = Path(external_dir);
+	}
+
+	// Might also add an option to move it to internal / non-visible storage, but there's
+	// little point, really.
+
 	g_Config.memStickDirectory = Path(external_dir);
 	g_Config.flash0Directory = Path(external_dir) / "flash0";
 
 	Path memstickDirFile = g_Config.internalDataDirectory / "memstick_dir.txt";
 	if (File::Exists(memstickDirFile)) {
 		std::string memstickDir;
-		File::ReadFileToString(true, memstickDirFile, memstickDir);
-		Path memstickPath(memstickDir);
-		if (!memstickPath.empty() && File::Exists(memstickPath)) {
-			g_Config.memStickDirectory = memstickPath;
-		} else {
-		    ERROR_LOG(SYSTEM, "Couldn't read directory '%s' specified by memstick_dir.txt.", memstickDir.c_str());
+		if (File::ReadFileToString(true, memstickDirFile, memstickDir)) {
+			Path memstickPath(memstickDir);
+			if (!memstickPath.empty() && File::Exists(memstickPath)) {
+				g_Config.memStickDirectory = memstickPath;
+				INFO_LOG(SYSTEM, "Memstick Directory from memstick_dir.txt: '%s'", g_Config.memStickDirectory.c_str());
+			} else {
+				ERROR_LOG(SYSTEM, "Couldn't read directory '%s' specified by memstick_dir.txt.", memstickDir.c_str());
+				if (System_GetPropertyBool(SYSPROP_ANDROID_SCOPED_STORAGE)) {
+					//
+				}
```

Comment:
Hm, what should we do here?  Maybe this is a place to REPORT?

-[Unknown]

Reply:
I'll get back to it before we enable.

Reply:
Should we reset it to `Path(external_dir)` or `g_Config.defaultCurrentDirectory` perhaps?  Better than not working or saving settings at all...

-[Unknown]

Reply:
Yeah, true. Although in the end I think we want to pop up the choose-your-memstick dialog in this case.